### ISSUE TITLE
nonzero() requires swap for float types (migrated from Trac #1662)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3213]
Original ticket http://projects.scipy.org/numpy/ticket/1662
Reported 2010-11-05 by atmention:mwiebe, assigned to unknown.

The current floating-point nonzero() implementation is incorrect for non-native byte order.  Swapping the order should have no effect, but it can change the result as visible here:

```
>>> import numpy as np
>>> a = np.array([0x80000000, 0x00000080, 0], dtype=np.uint32)
>>> a.dtype = np.float32
>>> print a.nonzero()
(array([1]),)
>>> print a.byteswap().newbyteorder().nonzero()
(array([0]),)
```

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2010-11-05

https://github.com/numpy/numpy/pull/10

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2010-11-07

Fixed in commit  6e5590ff95219c5323f3. Thanks.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-05-24
